tests/kernel/threads/usage/src/main.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @addtogroup t_threads_usage
+ * @{
+ * @defgroup t_threads_usage_api test_threads_lifecycle_api
+ * @brief TestPurpose: verify zephyr basic threads lifecycle apis
+ * @}
+ */
+
+#include <ztest.h>
+extern void test_thread1(void);
+extern void test_thread2(void);
+
+__kernel struct k_thread tdata;
+#define STACK_SIZE (2048 + CONFIG_TEST_EXTRA_STACKSIZE)
+K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+
```

Comment:
what is the use of the above thread strutcs and stack? any way you defining them again in task.c

Reply:
also do we need another file, we can just move the tests and test_main to a single file.